### PR TITLE
Fix gcc8 format-truncation error and md5 compile error

### PR DESCRIPTION
**Pull request by sztsian**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 hbackup_server: hbackup_server.c
-	gcc -o hbackup_server -g -Wall hbackup_server.c -lssl
+	gcc -o hbackup_server -g -Wall hbackup_server.c -lssl -lcrypto
```
Comment (zhsj) [inline Makefile]: -lcrypto is enough

**Comment by sztsian** [comment]
fyi

>$ gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/libexec/gcc/x86_64-redhat-linux/8/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-redhat-linux
Configured with: ../configure --enable-bootstrap --enable-languages=c,c++,fortran,objc,obj-c++,ada,go,lto --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=http://bugzilla.redhat.com/bugzilla --enable-shared --enable-threads=posix --enable-checking=release --enable-multilib --with-system-zlib --enable-__cxa_atexit --disable-libunwind-exceptions --enable-gnu-unique-object --enable-linker-build-id --with-gcc-major-version-only --with-linker-hash-style=gnu --enable-plugin --enable-initfini-array --with-isl --enable-libmpx --enable-offload-targets=nvptx-none --without-cuda-driver --enable-gnu-indirect-function --enable-cet --with-tune=generic --with-arch_32=i686 --build=x86_64-redhat-linux
Thread model: posix
gcc version 8.1.1 20180502 (Red Hat 8.1.1-1) (GCC)

File: Makefile (hunk 1 above)
Comment (sztsian) [reply Makefile]: fixed

File: hbackup_server.c [hunk 2]
```diff
@@ -445,7 +445,8 @@ int RecvHashedFile(int fd, char *md5sum, char *hashed_file_name, size_t file_len
 	check_and_create_dir(file_name);
 	FILE *fp = fopen(file_name, "w");
 	if (fp == NULL) {
-		snprintf(buf, MAXLEN, "ERROR open tmpfile %s for write\n", file_name);
+		const char *format = "ERROR open tmpfile %s for write\n";
```
Comment (zhsj) [inline hbackup_server.c]: `%.*s`

File: hbackup_server.c [hunk 3]
```diff
@@ -445,7 +445,8 @@ int RecvHashedFile(int fd, char *md5sum, char *hashed_file_name, size_t file_len
...
+		snprintf(buf, MAXLEN, format, MAXLEN-strlen(format), file_name);
```
Comment (zhsj) [inline hbackup_server.c]:
```
const char format[] = "ERROR open tmpfile %.*s for write\n";
snprintf(buf, MAXLEN, format, (int)(MAXLEN-sizeof(format)), file_name);
```

File: hbackup_server.c [hunk 4]
```diff
@@ -445,7 +445,8 @@ int RecvHashedFile(int fd, char *md5sum, char *hashed_file_name, size_t file_len
 	check_and_create_dir(file_name);
 	FILE *fp = fopen(file_name, "w");
 	if (fp == NULL) {
-		snprintf(buf, MAXLEN, "ERROR open tmpfile %s for write\n", file_name);
+		const char *format = "ERROR open tmpfile %.s for write\n";
+		snprintf(buf, MAXLEN, format, (int)(MAXLEN-sizeof(format)), file_name);
```
Comment (zhsj) [inline hbackup_server.c]: If you want to use `char *format`, you need to use `strlen(format)`

**Pull request merged**